### ISSUE TITLE
[P2 - Major] Comments absent from all scripts

### ISSUE THREAD

**Issue by vaiolis** [arnuvtandon/HomesteadFBLAv2#6]
None of the scripts I've reviewed so far have comments explaining the purpose of each function or descriptors of their parameters and return values. There are some inline comments for some individual code snippets, but that is not sufficient for providing reviewers with appropriate metadata to reach a high-level understanding of each script at a glance.

It would be very valuable for future contributors to have this information. [JSDoc](https://jsdoc.app/about-getting-started.html) is an example of a system for providing this information. To use `PushMaterialsSnapshot.js` as an example:

```
/**
 * This routine retrieves a snapshot of the materials collection and copies this data to a new document in the 
 * backups collection.
 * @param {boolean} isARevert - if truthy, designates this invocation as a revert operation
 */
function PushMaterialsSnapshot(isARevert) {
```

**Comment by arnuvtandon** [comment]
I agree with you that this is urgent, all the scripts should have descriptive commenting. Before we work on new features, I will put in a few hours to bring the documentation up to date.
